### PR TITLE
Outputs enhancement to require Description and SampleConfig functions

### PR DESCRIPTION
**Pull request by sparrc**
/cc @jipperinbham please verify the `SampleConfg()` and `Description()` functions for the `datadog` output, thanks!

### CODE REVIEW

File: config.go [hunk 1]
```diff
@@ -348,40 +346,28 @@ var header = `# Telegraf configuration
...
-# Connection timeout (for the connection with InfluxDB), formatted as a string.
-# Valid time units are "ns", "us" (or "µs"), "ms", "s", "m", "h".
-# If not provided, will default to 0 (no timeout)
-# timeout = "5s"
-
-# username = "telegraf"
-# password = "metricsmetricsmetricsmetrics"
+# Tags can also be specified via a normal map, but only one form at a time:
+[tags]
+	# dc = "us-east-1" }
```
Comment (jipperinbham) [inline config.go]: I think that closing bracket `}` is out of place/not needed

File: outputs/datadog/datadog.go [hunk 2]
```diff
@@ -91,6 +99,14 @@ func (d *Datadog) Write(bp client.BatchPoints) error {
 	return nil
 }
 
+func (d *Datadog) SampleConfig() string {
+	return sampleConfig
+}
+
+func (d *Datadog) Description() string {
+	return "Configuration for DataDog server to send metrics to."
```
Comment (jipperinbham) [inline outputs/datadog/datadog.go]: purely semantics but maybe s/server/API

**Comment by jipperinbham** [comment]
@sparrc couple comments but nothing ground breaking :) 

thanks for adding this!

**Comment by sparrc** [comment]
thanks @jipperinbham for the feedback :+1:

**Pull request closed without merging**